pkg/event/target/mqtt.go

```diff
@@ -216,17 +217,12 @@ func NewMQTTTarget(id string, args MQTTArgs, doneCh <-chan struct{}, loggerOnce
 			return nil, err
 		}
 
-		go retryRegister()
-
 		// Replays the events from the store.
 		eventKeyCh := replayEvents(target.store, doneCh, loggerOnce, target.ID())
 
 		// Start replaying events from the store.
 		go sendEvents(target, eventKeyCh, doneCh, loggerOnce)
-	} else {
```

Comment:
We have to return the connection error back. If the queueDir is not configured. The retry can happen only if queueDir is not nil.

Reply:
in any case queueDir will be set in most production deployments, so we are invalidating the TestNotificationTargets here @Praveenrajmani - first, we shouldn't have two ways to do the same thing. It's confusing and wrong.

It already made my testing harder that we have multiple ways to do target connect() so I would suggest we change every other target to do the same. Validation() of settings should happen regardless of queueDir is set or not. So I will keep this code as-is for now.